File: Apps/BankCore/backend/src/main/java/org/d3javu/backend/kafka/main/card/CardCreateRequest.java
package org.d3javu.backend.kafka.main.card;

import org.d3javu.backend.model.account.AccountType;
import org.d3javu.backend.model.card.CardType;

public record CardCreateRequest(
        Long clientId,
        String email,
        Long accountId,
        CardType type,
        String name
) {}

File: Apps/BankBase/backend/src/main/scala/internalkafka/messages/card/CardCreateRequest.scala
package internalkafka.messages.card

import model.card.CardType
import zio.json.{DeriveJsonCodec, JsonCodec}

case class CardCreateRequest(
                              clientId: Long,
                              email: String,
                              accountId: Long,
                              cardType: CardType,
                              name: String
                            )

object CardCreateRequest {
  inline given JsonCodec[CardCreateRequest] = DeriveJsonCodec.gen[CardCreateRequest]
}
